curvefs/src/metaserver/partition.cpp

```diff
@@ -57,14 +57,40 @@ MetaStatusCode Partition::CreateDentry(const Dentry& dentry) {
     if (!IsInodeBelongs(dentry.fsid(), dentry.parentinodeid())) {
         return MetaStatusCode::PARTITION_ID_MISSMATCH;
     }
-    return dentryManager_->CreateDentry(dentry);
+    MetaStatusCode ret = dentryManager_->CreateDentry(dentry);
+    if (MetaStatusCode::OK == ret) {
+        return inodeManager_->UpdateInodeWhenCreateSubNode(
+            dentry.fsid(), dentry.parentinodeid());
+    } else if (MetaStatusCode::IDEMPOTENCE_OK == ret) {
+        return MetaStatusCode::OK;
+    } else {
+        return ret;
+    }
+}
+
+MetaStatusCode Partition::InsertDentry(const Dentry& dentry) {
+    if (!IsInodeBelongs(dentry.fsid(), dentry.parentinodeid())) {
+        return MetaStatusCode::PARTITION_ID_MISSMATCH;
+    }
+    MetaStatusCode ret = dentryManager_->CreateDentry(dentry);
+    if (MetaStatusCode::IDEMPOTENCE_OK == ret) {
+        return MetaStatusCode::OK;
+    } else {
+        return ret;
+    }
```

Comment:
there is no much difference with `CreateDentry`, use another flag to control whether do update

Reply:
done